File: C#/Anagram/Anagram.cs
using System.Linq;
using System.Collections.Generic;

public class Anagram
{
    private string _baseWord;

    public Anagram(string baseWord)
    {
        _baseWord = baseWord;
    }

    private static string SortedString(string s)
    {
        return string.Join("", s.OrderBy(x => x).ToArray());
    }

    public string[] FindAnagrams(string[] potentialMatches)
    {
        List<string> matches = new();

        foreach (var potential in potentialMatches)
        {
            // Can't be an anagram if it already matches
            if(potential.ToLower() == _baseWord.ToLower())
            {
                continue;
            }

            else if (SortedString(potential.ToLower()) == SortedString(_baseWord.ToLower()))
            {
                matches.Add(potential);
            }
        }

        return matches.ToArray();
    }
}

File: JavaScript/anagram.js
const sortString = (str) => {
    return str.split('').sort().join('');
}

const findAnagrams = (baseword, potentials) => {
  var matches = [];

  potentials.forEach(potential => {
    var b = baseword.toLowerCase();
    var p = potential.toLowerCase();

    // Can't be an anagram if it already matches
    if(b != p) {       
      if(sortString(b) === sortString(p)) {
          matches.push(potential);
      }
    }
  });

  return matches;
};  

const inputs = [
    { baseword:'Orchestra', words:['cashregister', 'Carthorse', 'radishes'] },
    { baseword:'nose', words:['Eons', 'ONES'] },
    { baseword:'solemn', words:['lemons', 'cherry', 'melons'] },
];

inputs.forEach(input => {
    console.log(`${input.baseword} -> ${findAnagrams(input.baseword, input.words)}`);
});